### DISCUSSION TITLE
Loading configuration file from c++ API

### DISCUSSION THREAD

**Discussion by marcomontevechi1** [curl/curl#14577] [Q&A] [answered]
Hello, community. I'm using curl in a custom application and I want an elegant way of setting all curl options. I though about having a configuration file, but although i have seen that the [command-line utility can load options from a config file](https://everything.curl.dev/cmdline/configfile.html), I have not found any easy function to be used in my application that would do this.

I'm thinking about creating a program that does that but it really feels like I'm reinventing the wheel?

I'm looking for something that would probably look like this:

```
  CURL *curl = curl_easy_init();
  curl_load_config_file(curl, "/config/file/path");
```

And load_config_file parses the file and sets all curl options, possibly returning all options that were set and their values.

Thanks in advance for any help.

**Comment by bagder** [comment, marked as answer]
The loading and using of a config file is code in the command line tool only, it is not functionality provide by libcurl. Thus: there is no provided API for doing this.

**Reply by marcomontevechi1** [reply]
Understood, thanks tor the reply
